### ISSUE TITLE
Checkpoint and model name responding relationship

### ISSUE THREAD

**Issue by LittleByLittle2** [GeorgeSeif/Semantic-Segmentation-Suite#139]
Please fill out this issue template before submitting. **Issues which do not fill out this template, or are already answered in the FAQs will simply be closed.**

**Please go to Stack Overflow for help and support.** Also check past issues as many are repeats. Also check out the Frequently Asked Questions (FAQs) below in case your question has already been answered in an issue!

Issues should be one of the following:

1. Feature requests
2. Bug reports

------------------------

## Please specify the following information when submitting an issue:
- **What are your command line arguments?:**
python predict.py  --image       /home/yuu/Documents/Project/sceneflow/tests/00000604.png --checkpoint_path models/resnet_v2_101.ckpt --model DeepLabV3  --dataset CamVid
- **Have you written any custom code?**:
No
- **What have you done to try and solve this issue?:**
I tried different check_point path and model name, but all has the problem.
- **TensorFlow version?**:
1.5.0

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug or a feature request.

Some tensor name from the module is not in checkpoint. So is there any responding relationship between checkpoint and model name ?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

NotFoundError (see above for traceback): Tensor name "inception_v4/Mixed_3a/Branch_1/Conv2d_0a_3x3/BatchNorm/beta" not found in checkpoint files models/resnet_v2_101.ckpt
	 [[Node: save/RestoreV2_24 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2_24/tensor_names, save/RestoreV2_24/shape_and_slices)]]
	 [[Node: save/RestoreV2_560/_121 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_1344_save/RestoreV2_560", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

### FAQs

- **Question:** I got an `InvalidArgumentError` saying that `Dimensions of inputs should match` **Answer:** See issue #17

- **Question:** Can you upload pre-trained weights for these networks? **Answer:** See issue #57

- **Question:** Do I need a GPU to train these models? **Answer:** Technically no, but I'd highly recommend it. I was able to train the models pretty well in about a day using a 1080Ti GPU. Training on CPU would take much longer than that.

- **Question:** Will you be adding the FCN or U-Net models? **Answer:** No I won't be adding those simply because they're a few years old and state-of-the-art has moved past that.

- **Question:** I got an invalid argument error when using the InceptionV4 model. Am I doing something wrong? **Answer:** No you're not! Due to the design of the InceptiveV4 model, when you end up upsampling you do some rounding which creates a shape mismatch. _This only happens when you end up having to use the `end_points['pool5']`_. See the code for some of the models if you want to check whether the model will use `end_points['pool5']`.

**Comment by GeorgeSeif** [comment]
The checkpoint path you have provided is for a **classification** network used for feature extraction. If you want to run prediction for segmentation, you'll need to train a model first! :)
